### PR TITLE
Upgrade packages

### PR DESCRIPTION
**Pull request by kenjinp**
In order to remove security warnings in yarn, and to prepare for yarn 2 migration, I've updated our dependencies. 

the process went like this:
- `yarn upgrade`
- fix new type errors created by dependency updates
- `yarn style --write` (prettier was also updated, it seems)
- `yarn test -u` (update snaps)

### CODE REVIEW

File: src/shared/utils/download.ts [hunk 1]
```diff
@@ -1,15 +1,6 @@
...
-    window.navigator.msSaveBlob(blob, filename);
-  } else {
-    const elem = window.document.createElement('a');
-    elem.href = window.URL.createObjectURL(blob);
-    elem.download = filename;
-    document.body.appendChild(elem);
-    elem.click();
-    document.body.removeChild(elem);
-  }
+  window.navigator.msSaveBlob(blob, filename);
```
Comment (dhaneshnm) [inline src/shared/utils/download.ts]: Why did we remove the else block?

File: src/shared/App.tsx [hunk 2]
```diff
@@ -20,7 +21,8 @@ const App: React.FC = () => {
   // use it as the location for the <Switch> so
   // we show the gallery in the background, behind
   // the modal.
-  const background = location.state && location.state.background;
+  const background =
```
Comment (dhaneshnm) [inline src/shared/App.tsx]: Are there any changes in Location/history packages? 
location.state.background vs location.state as { background?: Location }).background (?)

Comment (kenjinp) [reply src/shared/App.tsx]: The history @types package was changed, yeah :)

File: src/shared/utils/download.ts (hunk 1 above)
Comment (kenjinp) [reply src/shared/utils/download.ts]: that `window.navigator.msSaveOrOpenBlob` function is guaranteed to exist

**Pull request merged**